completions/_umount.linux

```diff
@@ -25,9 +25,7 @@ _reply_compgen_array()
     ecur=${ecur//\'/\\\'}
 
     # Actually generate completions.
-    local ifs=$IFS
     IFS=$'\n' eval 'COMPREPLY=(`compgen -W "$wlist" -- "${ecur}"`)'
```

Comment:
In POSIX mode, the effect of the variable assignment before special builtins (including `eval`) remains after the execution of the command.

```bash
$ bash --norc
$ set -o posix
$ declare -p IFS
declare -- IFS="
"
$ IFS=: eval 'x=($SHELLOPTS)'
$ declare -p IFS x
declare -x IFS=":"
```

So we actually need to explicitly restore the original IFS. In this sense, the original code is a bit strange. I think we may rewrite it as

```suggestion
    local IFS=$'\n'
    COMPREPLY=($(compgen -W "$wlist" -- "${ecur}"))
    _comp_unlocal IFS
```

What do you think?

Reply:
I think I am learning a lot about shell! I will apply this suggestion.